File: frontend/src/api/recruitment.ts
import client from './client';

// ── Types ─────────────────────────────────────────────────────────────────────

export interface ManpowerRequisition {
  id: string;
  tenantId: string;
  requisitionNumber: string;
  departmentName: string;
  designationTitle: string;
  headCount: number;
  employmentType: string;
  priority: string;
  justification: string;
  requiredSkills: string;
  minExperienceYears: number | null;
  maxExperienceYears: number | null;
  budgetFrom: number | null;
  budgetTo: number | null;
  targetJoiningDate: string | null;
  status: string;
  requestedByName: string;
  rejectionReason: string;
  createdAtUtc: string;
  submittedAtUtc: string | null;
  approvedAtUtc: string | null;
}

export interface JobOpening {
  id: string;
  jobCode: string;
  requisitionId: string | null;
  title: string;
  departmentName: string;
  designationTitle: string;
  employmentType: string;
  headCount: number;
  filledCount: number;
  description: string;
  requirements: string;
  responsibilities: string;
  salaryFrom: number | null;
  salaryTo: number | null;
  location: string;
  status: string;
  assignedHrName: string;
  createdAtUtc: string;
  activeApplications?: number;
  remaining?: number;
}

export interface Candidate {
  id: string;
  firstName: string;
  lastName: string;
  fullName: string;
  email: string;
  phone: string;
  currentJobTitle: string;
  currentCompany: string;
  totalExperienceYears: number;
  educationLevel: string;
  nationality: string;
  linkedInUrl: string;
  source: string;
  status: string;
  tags: string;
  createdAtUtc: string;
  applicationCount?: number;
}

export interface JobApplication {
  id: string;
  jobOpeningId: string;
  jobTitle: string;
  candidateId: string;
  candidateName: string;
  candidateEmail: string;
  stage: string;
  stageOrder: number;
  status: string;
  rejectionReason: string;
  offeredSalary: number | null;
  appliedAtUtc: string;
  stageChangedAtUtc: string | null;
  hiredAtUtc: string | null;
  onboardingDraftId: string | null;
}

export interface ApplicationEvent {
  id: string;
  eventType: string;
  stage: string;
  notes: string;
  performedByName: string;
  createdAtUtc: string;
}

export interface InterviewSchedule {
  id: string;
  applicationId: string;
  interviewType: string;
  interviewerNames: string;
  scheduledAt: string;
  durationMinutes: number;
  mode: string;
  meetingLink: string;
  location: string;
  status: string;
  overallRating: number | null;
  recommendation: string;
  feedbackNotes: string;
  completedAt: string | null;
  createdAtUtc: string;
}

export interface OfferLetter {
  id: string;
  applicationId: string;
  candidateName: string;
  offeredJobTitle: string;
  offeredDepartment: string;
  startDate: string;
  basicSalary: number;
  housingAllowance: number;
  transportAllowance: number;
  otherAllowances: number;
  grossSalary: number;
  probationMonths: number;
  status: string;
  generatedAtUtc: string;
  sentAtUtc: string | null;
  responseDeadline: string | null;
  acceptedAtUtc: string | null;
}

export interface KanbanStage {
  stage: string;
  order: number;
  applications: JobApplication[];
}

export interface ApplicationDetail {
  application: JobApplication;
  candidate: Candidate | null;
  events: ApplicationEvent[];
  interviews: InterviewSchedule[];
  offer: OfferLetter | null;
}

export interface RecruitmentStats {
  openPositions: number;
  totalOpenings: number;
  activeApplications: number;
  hiredThisMonth: number;
  offersPending: number;
}

export interface RequisitionStats {
  total: number;
  draft: number;
  pending: number;
  approved: number;
  converted: number;
}

// ── API ───────────────────────────────────────────────────────────────────────

export const requisitionsApi = {
  list: (params: { status?: string; priority?: string; page?: number; pageSize?: number } = {}) =>
    client.get<{ items: ManpowerRequisition[]; total: number }>('/api/recruitment/requisitions', { params }).then(r => r.data),

  get: (id: string) =>
    client.get<ManpowerRequisition>(`/api/recruitment/requisitions/${id}`).then(r => r.data),

  create: (body: {
    departmentId?: string; departmentName: string; designationId?: string; designationTitle: string;
    headCount: number; employmentType: string; priority: string; justification: string;
    requiredSkills: string; minExperienceYears?: number; maxExperienceYears?: number;
    budgetFrom?: number; budgetTo?: number; targetJoiningDate?: string; requestedByName: string;
  }) => client.post<ManpowerRequisition>('/api/recruitment/requisitions', body).then(r => r.data),

  submit: (id: string) =>
    client.post<ManpowerRequisition>(`/api/recruitment/requisitions/${id}/submit`).then(r => r.data),

  approve: (id: string) =>
    client.post<ManpowerRequisition>(`/api/recruitment/requisitions/${id}/approve`, {}).then(r => r.data),

  reject: (id: string, reason: string) =>
    client.post<ManpowerRequisition>(`/api/recruitment/requisitions/${id}/reject`, { reason }).then(r => r.data),

  stats: () =>
    client.get<RequisitionStats>('/api/recruitment/requisitions/stats').then(r => r.data),
};

export const openingsApi = {
  list: (params: { status?: string; page?: number } = {}) =>
    client.get<{ items: JobOpening[]; total: number }>('/api/recruitment/openings', { params }).then(r => r.data),

  get: (id: string) =>
    client.get<{ opening: JobOpening; stageCounts: { stage: string; count: number }[] }>(`/api/recruitment/openings/${id}`).then(r => r.data),

  create: (body: {
    requisitionId?: string; title: string; departmentName: string; designationTitle: string;
    employmentType: string; headCount: number; description: string; requirements: string;
    responsibilities: string; salaryFrom?: number; salaryTo?: number; location: string; assignedHrName: string;
  }) => client.post<JobOpening>('/api/recruitment/openings', body).then(r => r.data),

  close: (id: string) =>
    client.post<JobOpening>(`/api/recruitment/openings/${id}/close`).then(r => r.data),

  stats: () =>
    client.get<RecruitmentStats>('/api/recruitment/openings/stats').then(r => r.data),
};

export const candidatesApi = {
  list: (params: { search?: string; status?: string; page?: number; pageSize?: number } = {}) =>
    client.get<{ items: Candidate[]; total: number }>('/api/recruitment/candidates', { params }).then(r => r.data),

  create: (body: {
    firstName: string; lastName: string; email: string; phone: string;
    currentJobTitle: string; currentCompany: string; totalExperienceYears: number;
    educationLevel: string; nationality: string; linkedInUrl: string; source: string; tags: string;
  }) => client.post<Candidate>('/api/recruitment/candidates', body).then(r => r.data),
};

// ── Extended Types ────────────────────────────────────────────────────────────

export interface WorkforcePlan {
  id: string;
  planCode: string;
  planYear: number;
  planName: string;
  departmentName: string;
  currentHeadcount: number;
  plannedHeadcount: number;
  gapCount: number;
  budgetAllocated: number;
  budgetUtilized: number;
  currencyCode: string;
  status: string;
  notes: string;
  createdByName: string;
  createdAtUtc: string;
  approvedAtUtc: string | null;
}

export interface InterviewFeedback {
  id: string;
  interviewScheduleId: string;
  applicationId: string;
  interviewerName: string;
  interviewerRole: string;
  communicationScore: number;
  technicalScore: number;
  cultureFitScore: number;
  problemSolvingScore: number;
  leadershipScore: number;
  overallScore: number;
  strengths: string;
  concerns: string;
  notes: string;
  recommendation: string;
  submittedAtUtc: string;
}

export interface AssessmentTemplate {
  id: string;
  code: string;
  title: string;
  description: string;
  assessmentType: string;
  durationMinutes: number;
  passingScore: number;
  totalMarks: number;
  isRandomized: boolean;
  audience: string;
  isActive: boolean;
  createdAtUtc: string;
}

export interface CandidateAssessment {
  id: string;
  applicationId: string;
  candidateId: string;
  templateId: string;
  templateName: string;
  status: string;
  sentAtUtc: string | null;
  completedAtUtc: string | null;
  expiresAtUtc: string | null;
  scoreObtained: number | null;
  totalMarks: number | null;
  scorePercentage: number | null;
  passed: boolean | null;
  createdAtUtc: string;
}

export interface OfferApproval {
  id: string;
  offerLetterId: string;
  stepOrder: number;
  approverName: string;
  approverRole: string;
  status: string;
  comments: string;
  decidedAtUtc: string | null;
}

export interface OnboardingChecklist {
  id: string;
  code: string;
  name: string;
  description: string;
  applicableTo: string;
  departmentName: string;
  isActive: boolean;
}

export interface OnboardingTask {
  id: string;
  checklistId: string | null;
  employeeId: string | null;
  applicationId: string | null;
  taskTitle: string;
  taskDescription: string;
  category: string;
  assignedToName: string;
  status: string;
  dueDate: string | null;
  completedDate: string | null;
  notes: string;
  orderIndex: number;
  isMandatory: boolean;
  createdAtUtc: string;
}

// ── Extended API Clients ───────────────────────────────────────────────────────

export const workforcePlanningApi = {
  list: (year?: number, status?: string) =>
    client.get<{ total: number; items: WorkforcePlan[] }>('/api/recruitment/workforce-planning', { params: { year, status } }).then(r => r.data),

  get: (id: string) =>
    client.get<WorkforcePlan>(`/api/recruitment/workforce-planning/${id}`).then(r => r.data),

  create: (body: { planName: string; planYear: number; departmentId?: string; departmentName?: string; currentHeadcount: number; plannedHeadcount: number; budgetAllocated: number; currencyCode?: string; notes?: string }) =>
    client.post<WorkforcePlan>('/api/recruitment/workforce-planning', body).then(r => r.data),

  updateStatus: (id: string, status: string) =>
    client.patch<WorkforcePlan>(`/api/recruitment/workforce-planning/${id}/status`, { status }).then(r => r.data),

  summary: (year?: number) =>
    client.get<{ totalPlans: number; draft: number; approved: number; inProgress: number; totalGap: number; totalBudget: number }>('/api/recruitment/workforce-planning/summary', { params: { year } }).then(r => r.data),
};

export const interviewsApi = {
  list: (applicationId?: string, status?: string, page = 1) =>
    client.get<{ total: number; items: InterviewSchedule[] }>('/api/recruitment/interviews', { params: { applicationId, status, page } }).then(r => r.data),

  get: (id: string) =>
    client.get<{ interview: InterviewSchedule; feedbacks: InterviewFeedback[] }>(`/api/recruitment/interviews/${id}`).then(r => r.data),

  schedule: (body: { applicationId: string; interviewType: string; interviewerNames: string; scheduledAt: string; durationMinutes: number; mode: string; meetingLink?: string; location?: string }) =>
    client.post<InterviewSchedule>('/api/recruitment/interviews', body).then(r => r.data),

  complete: (id: string, body: { overallRating: number; recommendation: string; feedbackNotes?: string }) =>
    client.patch<InterviewSchedule>(`/api/recruitment/interviews/${id}/complete`, body).then(r => r.data),

  cancel: (id: string) =>
    client.patch<InterviewSchedule>(`/api/recruitment/interviews/${id}/cancel`, {}).then(r => r.data),

  submitFeedback: (id: string, body: { interviewerName?: string; interviewerRole?: string; communicationScore: number; technicalScore: number; cultureFitScore: number; problemSolvingScore: number; leadershipScore: number; overallScore: number; strengths?: string; concerns?: string; notes?: string; recommendation: string }) =>
    client.post<InterviewFeedback>(`/api/recruitment/interviews/${id}/feedback`, body).then(r => r.data),

  getFeedback: (id: string) =>
    client.get<InterviewFeedback[]>(`/api/recruitment/interviews/${id}/feedback`).then(r => r.data),
};

export const assessmentsApi = {
  listTemplates: () =>
    client.get<AssessmentTemplate[]>('/api/recruitment/assessments/templates').then(r => r.data),

  getTemplate: (id: string) =>
    client.get<{ template: AssessmentTemplate; questions: unknown[] }>(`/api/recruitment/assessments/templates/${id}`).then(r => r.data),

  createTemplate: (body: { code: string; title: string; description?: string; assessmentType: string; durationMinutes: number; passingScore: number; isRandomized: boolean; audience?: string }) =>
    client.post<AssessmentTemplate>('/api/recruitment/assessments/templates', body).then(r => r.data),

  list: (applicationId?: string, status?: string) =>
    client.get<{ total: number; items: CandidateAssessment[] }>('/api/recruitment/assessments', { params: { applicationId, status } }).then(r => r.data),

  send: (body: { applicationId: string; templateId: string; expiryDays?: number }) =>
    client.post<CandidateAssessment>('/api/recruitment/assessments/send', body).then(r => r.data),

  recordResult: (id: string, scoreObtained: number) =>
    client.patch<CandidateAssessment>(`/api/recruitment/assessments/${id}/result`, { scoreObtained }).then(r => r.data),
};

export const offersApi = {
  list: (applicationId?: string, status?: string) =>
    client.get<{ total: number; items: OfferLetter[] }>('/api/recruitment/offers', { params: { applicationId, status } }).then(r => r.data),

  get: (id: string) =>
    client.get<{ offer: OfferLetter; approvals: OfferApproval[] }>(`/api/recruitment/offers/${id}`).then(r => r.data),

  create: (body: { applicationId: string; offeredJobTitle: string; offeredDepartment?: string; startDate: string; basicSalary: number; housingAllowance: number; transportAllowance: number; otherAllowances: number; probationMonths: number; contentHtml?: string; responseDeadline?: string }) =>
    client.post<OfferLetter>('/api/recruitment/offers', body).then(r => r.data),

  send: (id: string) =>
    client.patch<OfferLetter>(`/api/recruitment/offers/${id}/send`, {}).then(r => r.data),

  accept: (id: string) =>
    client.patch<OfferLetter>(`/api/recruitment/offers/${id}/accept`, {}).then(r => r.data),

  decline: (id: string, reason: string) =>
    client.patch<OfferLetter>(`/api/recruitment/offers/${id}/decline`, { reason }).then(r => r.data),
};

export const onboardingApi = {
  listChecklists: () =>
    client.get<OnboardingChecklist[]>('/api/recruitment/onboarding/checklists').then(r => r.data),

  listTasks: (params: { employeeId?: string; applicationId?: string; status?: string; page?: number } = {}) =>
    client.get<{ total: number; items: OnboardingTask[] }>('/api/recruitment/onboarding/tasks', { params }).then(r => r.data),

  createTask: (body: { taskTitle: string; taskDescription?: string; category?: string; checklistId?: string; employeeId?: string; applicationId?: string; assignedToName?: string; dueDate?: string; isMandatory: boolean }) =>
    client.post<OnboardingTask>('/api/recruitment/onboarding/tasks', body).then(r => r.data),

  updateStatus: (id: string, status: string, notes?: string) =>
    client.patch<OnboardingTask>(`/api/recruitment/onboarding/tasks/${id}/status`, { status, notes }).then(r => r.data),

  summary: (employeeId?: string, applicationId?: string) =>
    client.get<{ total: number; pending: number; inProgress: number; completed: number; blocked: number; mandatory: number; mandatoryCompleted: number; completionPct: number }>('/api/recruitment/onboarding/summary', { params: { employeeId, applicationId } }).then(r => r.data),
};

export const recruitmentReportsApi = {
  pipelineSummary: () =>
    client.get('/api/recruitment/reports/pipeline-summary').then(r => r.data),

  timeToHire: (year?: number, month?: number) =>
    client.get('/api/recruitment/reports/time-to-hire', { params: { year, month } }).then(r => r.data),

  sourceEffectiveness: () =>
    client.get('/api/recruitment/reports/source-effectiveness').then(r => r.data),

  openPositions: () =>
    client.get('/api/recruitment/reports/open-positions').then(r => r.data),

  aiInsights: () =>
    client.get('/api/recruitment/reports/ai-insights').then(r => r.data),
};

export const applicationsApi = {
  kanban: (jobOpeningId: string) =>
    client.get<{ stages: KanbanStage[]; rejected: JobApplication[] }>(`/api/recruitment/applications/kanban/${jobOpeningId}`).then(r => r.data),

  get: (id: string) =>
    client.get<ApplicationDetail>(`/api/recruitment/applications/${id}`).then(r => r.data),

  apply: (jobOpeningId: string, candidateId: string, notes?: string) =>
    client.post<JobApplication>('/api/recruitment/applications', { jobOpeningId, candidateId, notes }).then(r => r.data),

  advance: (id: string, notes?: string) =>
    client.post<JobApplication>(`/api/recruitment/applications/${id}/advance`, { notes }).then(r => r.data),

  reject: (id: string, reason: string) =>
    client.post<JobApplication>(`/api/recruitment/applications/${id}/reject`, { reason }).then(r => r.data),

  addNote: (id: string, notes: string) =>
    client.post(`/api/recruitment/applications/${id}/notes`, { notes }).then(r => r.data),

  scheduleInterview: (id: string, body: {
    interviewType: string; interviewerNames: string; scheduledAt: string;
    durationMinutes: number; mode: string; meetingLink: string; location: string;
  }) => client.post<InterviewSchedule>(`/api/recruitment/applications/${id}/interviews`, body).then(r => r.data),

  recordFeedback: (interviewId: string, body: { overallRating: number; recommendation: string; feedbackNotes: string }) =>
    client.post<InterviewSchedule>(`/api/recruitment/applications/interviews/${interviewId}/feedback`, body).then(r => r.data),

  generateOffer: (id: string, body: {
    department: string; startDate: string; basicSalary: number;
    housingAllowance: number; transportAllowance: number; otherAllowances: number; probationMonths: number;
  }) => client.post<OfferLetter>(`/api/recruitment/applications/${id}/offer`, body).then(r => r.data),

  sendOffer: (offerId: string) =>
    client.post<OfferLetter>(`/api/recruitment/applications/offers/${offerId}/send`).then(r => r.data),

  acceptOffer: (offerId: string) =>
    client.post(`/api/recruitment/applications/offers/${offerId}/accept`).then(r => r.data),

  declineOffer: (offerId: string, reason: string) =>
    client.post(`/api/recruitment/applications/offers/${offerId}/decline`, { reason }).then(r => r.data),

  getOfferHtml: (offerId: string) =>
    client.get<string>(`/api/recruitment/applications/offers/${offerId}/html`, { responseType: 'text' }).then(r => r.data),
};

File: backend-dotnet/Zayra.Api/Models/Recruitment.cs
using Zayra.Api.Domain.Entities;
namespace Zayra.Api.Models;

// ── Stage constants ────────────────────────────────────────────────────────────

public static class RecruitmentStages
{
    public static readonly (string Name, int Order)[] Pipeline =
    [
        ("Applied",    1),
        ("Screening",  2),
        ("Assessment", 3),
        ("Interview",  4),
        ("Offer",      5),
        ("Hired",      6),
    ];

    public static int OrderOf(string stage) =>
        Pipeline.FirstOrDefault(s => s.Name == stage).Order;

    public static string? Next(string stage)
    {
        var idx = Array.FindIndex(Pipeline, s => s.Name == stage);
        return idx >= 0 && idx < Pipeline.Length - 1 ? Pipeline[idx + 1].Name : null;
    }
}

// ── Manpower Requisition ───────────────────────────────────────────────────────

public class ManpowerRequisition : ITenantOwned
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid TenantId { get; set; }
    public string RequisitionNumber { get; set; } = string.Empty;       // MRQ-2026-0001
    public Guid? DepartmentId { get; set; }
    public string DepartmentName { get; set; } = string.Empty;
    public Guid? DesignationId { get; set; }
    public string DesignationTitle { get; set; } = string.Empty;
    public int HeadCount { get; set; } = 1;
    public string EmploymentType { get; set; } = "Full-Time";           // Full-Time/Part-Time/Contract/Internship
    public string Priority { get; set; } = "Medium";                    // Low/Medium/High/Critical
    public string Justification { get; set; } = string.Empty;
    public string RequiredSkills { get; set; } = string.Empty;
    public int? MinExperienceYears { get; set; }
    public int? MaxExperienceYears { get; set; }
    public decimal? BudgetFrom { get; set; }
    public decimal? BudgetTo { get; set; }
    public DateOnly? TargetJoiningDate { get; set; }
    public string Status { get; set; } = "Draft";                       // Draft/Submitted/PendingApproval/Approved/Rejected/Converted
    public Guid? RequestedByUserId { get; set; }
    public string RequestedByName { get; set; } = string.Empty;
    public int? RequestedByEmployeeId { get; set; }
    public string RejectionReason { get; set; } = string.Empty;
    public Guid? ApprovalRequestId { get; set; }
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? SubmittedAtUtc { get; set; }
    public DateTime? ApprovedAtUtc { get; set; }
    public DateTime? RejectedAtUtc { get; set; }
}

// ── Job Opening ────────────────────────────────────────────────────────────────

public class JobOpening : ITenantOwned
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid TenantId { get; set; }
    public string JobCode { get; set; } = string.Empty;                 // JOB-2026-0001
    public Guid? RequisitionId { get; set; }
    public string Title { get; set; } = string.Empty;
    public Guid? DepartmentId { get; set; }
    public string DepartmentName { get; set; } = string.Empty;
    public Guid? DesignationId { get; set; }
    public string DesignationTitle { get; set; } = string.Empty;
    public string EmploymentType { get; set; } = "Full-Time";
    public int HeadCount { get; set; } = 1;
    public int FilledCount { get; set; }
    public string Description { get; set; } = string.Empty;
    public string Requirements { get; set; } = string.Empty;           // newline-separated
    public string Responsibilities { get; set; } = string.Empty;
    public decimal? SalaryFrom { get; set; }
    public decimal? SalaryTo { get; set; }
    public string Location { get; set; } = string.Empty;
    public string Status { get; set; } = "Open";                       // Open/InProgress/OnHold/Closed/Cancelled
    public Guid? AssignedHrUserId { get; set; }
    public string AssignedHrName { get; set; } = string.Empty;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? PublishedAtUtc { get; set; }
    public DateTime? ClosedAtUtc { get; set; }
}

// ── Candidate (Talent Pool) ────────────────────────────────────────────────────

public class Candidate : ITenantOwned
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid TenantId { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string CurrentJobTitle { get; set; } = string.Empty;
    public string CurrentCompany { get; set; } = string.Empty;
    public decimal TotalExperienceYears { get; set; }
    public string EducationLevel { get; set; } = string.Empty;         // Diploma/Bachelor/Master/PhD
    public string Nationality { get; set; } = string.Empty;
    public string LinkedInUrl { get; set; } = string.Empty;
    public string ResumeUrl { get; set; } = string.Empty;
    public string Source { get; set; } = "Direct";                     // LinkedIn/Referral/JobBoard/Walk-In/Agency/Direct
    public string Status { get; set; } = "Active";                     // Active/Blacklisted
    public string Tags { get; set; } = string.Empty;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAtUtc { get; set; }
}

// ── Job Application (Pipeline record) ─────────────────────────────────────────

public class JobApplication : ITenantOwned
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid TenantId { get; set; }
    public Guid JobOpeningId { get; set; }
    public string JobTitle { get; set; } = string.Empty;
    public Guid CandidateId { get; set; }
    public string CandidateName { get; set; } = string.Empty;
    public string CandidateEmail { get; set; } = string.Empty;
    public string Stage { get; set; } = "Applied";
    public int StageOrder { get; set; } = 1;
    public string Status { get; set; } = "Active";                     // Active/Rejected/Withdrawn/Hired
    public string RejectionReason { get; set; } = string.Empty;
    public decimal? OfferedSalary { get; set; }
    public DateTime AppliedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? StageChangedAtUtc { get; set; }
    public DateTime? HiredAtUtc { get; set; }
    public Guid? OnboardingDraftId { get; set; }
}

// ── Application Event (Timeline) ──────────────────────────────────────────────

public class ApplicationEvent : ITenantOwned
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid TenantId { get; set; }
    public Guid ApplicationId { get; set; }
    public string EventType { get; set; } = string.Empty;              // StageAdvanced/Rejected/NoteAdded/InterviewScheduled/FeedbackRecorded/OfferGenerated/OfferSent/OfferAccepted/OfferDeclined/OnboardingStarted
    public string Stage { get; set; } = string.Empty;
    public string Notes { get; set; } = string.Empty;
    public Guid? PerformedByUserId { get; set; }
    public string PerformedByName { get; set; } = string.Empty;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
}

// ── Interview Schedule ─────────────────────────────────────────────────────────

public class InterviewSchedule : ITenantOwned
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid TenantId { get; set; }
    public Guid ApplicationId { get; set; }
    public string InterviewType { get; set; } = "HR Screening";        // HR Screening/Technical/Manager/Panel/Final
    public string InterviewerNames { get; set; } = string.Empty;
    public DateTime ScheduledAt { get; set; }
    public int DurationMinutes { get; set; } = 60;
    public string Mode { get; set; } = "Video";                        // InPerson/Video/Phone
    public string MeetingLink { get; set; } = string.Empty;
    public string Location { get; set; } = string.Empty;
    public string Status { get; set; } = "Scheduled";                  // Scheduled/Completed/Cancelled/NoShow
    public int? OverallRating { get; set; }                            // 1-5
    public string Recommendation { get; set; } = string.Empty;         // Proceed/Reject/Hold
    public string FeedbackNotes { get; set; } = string.Empty;
    public DateTime? CompletedAt { get; set; }
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
}

// ── Offer Letter ───────────────────────────────────────────────────────────────

public class OfferLetter : ITenantOwned
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid TenantId { get; set; }
    public Guid ApplicationId { get; set; }
    public string CandidateName { get; set; } = string.Empty;
    public string OfferedJobTitle { get; set; } = string.Empty;
    public string OfferedDepartment { get; set; } = string.Empty;
    public DateOnly StartDate { get; set; }
    public decimal BasicSalary { get; set; }
    public decimal HousingAllowance { get; set; }
    public decimal TransportAllowance { get; set; }
    public decimal OtherAllowances { get; set; }
    public decimal GrossSalary { get; set; }
    public int ProbationMonths { get; set; } = 3;
    public string ContentHtml { get; set; } = string.Empty;
    public string Status { get; set; } = "Draft";                      // Draft/Approved/Sent/Accepted/Declined/Expired
    public DateTime GeneratedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? SentAtUtc { get; set; }
    public DateTime? ResponseDeadline { get; set; }
    public DateTime? AcceptedAtUtc { get; set; }
    public DateTime? DeclinedAtUtc { get; set; }
    public string DeclineReason { get; set; } = string.Empty;
}
